app/deliveryoptions/delivery_options.py

```diff
@@ -0,0 +1,318 @@
+
+    Returns:
+        Processed content (str or List) based on the builder type
+    """
+    # Handle special case for distressing content
+    dcs_flag = False
+    if builder_type == "description" and has_distressing_content_match(
+        record_data.reference_number
+    ):
+        dcs_flag = True
+
+    # Default HTML building
+    if builder_type == "orderbuttons" and isinstance(
+        delivery_option_data, list
+    ):
+        result = []
+        for item in delivery_option_data:
+            processed_item = {}
+            for key, value in item.items():
+                if key in ["href", "text"]:
+                    processed_item[key] = html_builder(
+                        value,
+                        record_data,
+                        api_surrogate_data=api_surrogate_data,
+                    )
+                else:
+                    processed_item[key] = value
+            result.append(processed_item)
+        return result
+
+    # Standard HTML building
+    return html_builder(
+        delivery_option_data,
+        record_data,
+        api_surrogate_data=api_surrogate_data,
+        dcs=dcs_flag,
+    )
+
+
+def construct_delivery_options(
+    api_result: List, record: Record, request: HttpRequest
+) -> Dict[str, Any]:
+    """
+    Construct delivery options based on record and request information.
+
+    This is the main function called from records.py to build the delivery options
+    for a record.
+
+    Args:
+        api_result: List of delivery options
+        record: The record object
+        request: The HTTP request
+
+    Returns:
+        The constructed delivery options
+    """
+
+    if api_length := len(api_result) > 1:
+        raise ValueError(
+            f"Too many results ({api_length}) from DORIS database for IAID {record.iaid}"
+        )
+
+    delivery_options_context_dict = {}
+
+    reader_type = get_reader_type(request)
+
+    delivery_options_context_dict["reader_type"] = reader_type
+
+    do_dict = read_delivery_options(settings.DELIVERY_OPTIONS_CONFIG)
+
+    # Surrogate links is always present as a list, which can be empty
+    do_surrogate = surrogate_link_builder(api_result[0]["surrogateLinks"])
+
+    if (
+        api_result[0]["options"]
+        == AvailabilityCondition.ClosedRetainedDeptKnown
+    ):
+        # Special case. Sometimes, for record type 14 (ClosedRetainedDeptKnown), the department name does not match
+        # any entry in the DEPARTMENT_DETAILS dictionary. This shouldn't happen but it does. Therefore, reset the type
+        # with that for AvailabilityCondition.ClosedRetainedDeptUnKnown
+        if not get_dept(record.reference_number, "deptname"):
+            api_result[0][
+                "options"
+            ] = AvailabilityCondition.ClosedRetainedDeptUnKnown
+
+    # Get the specific delivery option for this artefact
+    delivery_option = get_delivery_option_dict(
+        do_dict, api_result[0]["options"]
+    )
```

Comment:
```suggestion
    delivery_option = do_dict["deliveryOptions"]["option"][api_result_options]
```
More clearer - this line tells the `api_result_options` is used to get the `do_dict` delivery option value. 
`get_delivery_option_dict()` should be redundant for the core code.

Reply:
Again, this is going in the next PR

Reply:
> Again, this is going in the next PR

#TODO comment should make it useful to know in the code.

Reply:
TODO's added